### PR TITLE
Refactor goal specification, clean up utils.py defaults, upgrade compatibility to newest brax/mjx

### PR DESCRIPTION
**Pull request by SimpleGeometry**
Preparatory PR -- new 7DoF arm environments coming in a separate PR after this merge is edited/approved.

Goal specification:
- Environments previously use contiguous subsequences for goal indices, i.e. goal_indices = [goal_start_idx, goal_end_idx). Sometimes these indices are difficult to merge together, and this PR refactors them into a more flexible definition of goal_indices (arbitrary indices).
- All environments have goal_indices specified, and references to goal start/end indices are replayed with goal_indices

Cleaning up utils.py defaults:
- Goal indices moved to be within the environment, rather than in the config. This seems to be more natural than defining it separately in utils.py
- Similarly, obs_dim is moved within the environment rather than config; corresponding calls requiring environments have been replaced.
- create_env overrides default __init__ arguments in some environments with, effectively, new defaults; they have been moved from create_env directly into the defaults in __init__.

Upgraded compatibility for brax/mjx:
- Updated environment.yml; moved brax from conda-forge to pip because newest version is not in conda-forge
- env.sys.dt replacements now swapped with tree_replace with env.sys.opt.timestep (brax 0.10.4 requirement)
- Newest mjx version additionally supports tendons, implicitfast integrator, etc.

Changes were tested with Ant and Humanoid, and seem to preserve performance parity + no errors.

### CODE REVIEW

File: training.py [hunk 1]
```diff
@@ -165,7 +166,7 @@ def progress(num_steps, metrics):
         group=args.group_name,
         name=args.exp_name,
         config=vars(args),
-        mode="online" if args.log_wandb else "disabled",
+        mode=mode="online" if args.log_wandb else "disabled",
```
Comment (wladekpal) [inline training.py:169]: I think that's a typo

File: training.py [hunk 2]
```diff
@@ -55,6 +55,7 @@ def main(args):
         pickle.dump(args, f)
 
     if args.use_traj_idx_wrapper:
+        raise Exception("traj_index_wrapper_factory does not yet support the new goal indexing scheme.")
```
Comment (wladekpal) [inline training.py:58]: In the future we will probably want remove trajectory index wrapper, it was made only for debugging of replay buffer, and now it only makes code complicated.

**Review by wladekpal — COMMENTED** [review]
One small typo, other than that lgtm.

File: training.py (hunk 1 above)
Comment (SimpleGeometry) [reply training.py]: Oops, yeah, fixed. I usually run jobs with mode="offline" and reverted the change incorrectly after testing. Test results should still be fine though.

**Pull request merged**